### PR TITLE
Sitemap and item config parsing adjustments

### PR DESCRIPTION
**Pull request by mherwege**
- Adapt sitemap and item lexers to changes in icon name syntax
- Restrict the elements that can be added to a sitemap
- Added extra sitemap validations (in line with xtext validation in core)
- Added test for sitemap parser and validation

This solves the issue in main UI created by https://github.com/openhab/openhab-core/pull/3539 and https://github.com/openhab/openhab-core/pull/3378.

### CODE REVIEW

**Review by florian-h05 — CHANGES_REQUESTED** [review]
I have tried to add Items from textual definition with the example from the core PR, however the icon value is not recognized and therefore not set. It seems that the functions to process the value are missing in items-lexer.nearley as well as for sitemap-lexer.nearley.

**Comment by lolodomo** [comment]
Look at #1839. Would be great if it could be fixed before tomorrow.

**Comment by florian-h05** [comment]
#1839 is a different issue, the problem with the Item lexer is that whatever you type as icon name is not saved to the Item, because the lexer does not work properly.

**Comment by mherwege** [comment]
I will need to check again. I thought it was working for me with the examples from the core PR and this this PR.  If it is not for you, there may still be something wrong. Unfortunately, I will not be able to do it before tomorrow night. I don’t have access to my PC at the moment.

**Comment by J-N-K** [comment]
Is this PR only needed to make the new feature work or is it needed to fix something?

**Comment by lolodomo** [comment]
> Is this PR only needed to make the new feature work or is it needed to fix something?

Probably only if you want to import items containing the extended syntax with several segments for the category/icon ?

**Comment by florian-h05** [comment]
> Is this PR only needed to make the new feature work or is it needed to fix something?

It seems it is only needed to to allow the new icon syntax in the „Add Items from textual definition“ and Sitemap editor’s codetab.

**Comment by J-N-K** [comment]
Ok, so it‘s not blocking M2.

**Comment by florian-h05** [comment]
BTW, I‘ve just checked what the transformation editor in the UI does after the recent core changes. It needs some adjustment to be fully operational, but IMO it‘s non-blocking since nothing really broke. I‘ve already worked a bit on the adjustment, should be able to open a PR soon.

**Comment by mherwege** [comment]
I know it is a bit late for the milestone, but I committed a fix to the parser.
Note that it also contains a fix for the syntax of an identifier in the items parser (bringing it in line with the one in the sitemap parser). The items parser was accepting more than it should have. This wasn't a real issue as long as the provided items file was valid, but would also import with invalid identifiers.

**Comment by florian-h05** [comment]
@ghys Feel free to take over the review, I won’t have the time to finish it anytime soon.

File: bundles/org.openhab.ui/web/src/pages/settings/pages/sitemap/__tests__/sitemap-edit_jest.spec.js [hunk 1]
```diff
@@ -186,39 +350,4 @@ describe('SitemapEdit', () => {
     wrapper.vm.validateWidgets()
     expect(lastDialogConfig).toBeFalsy()
   })
-
-  it('validates valuecolor', async () => {
```
Comment (florian-h05) [inline bundles/org.openhab.ui/web/src/pages/settings/pages/sitemap/__tests__/sitemap-edit_jest.spec.js:190]: Why did you remove that test?

**Review by florian-h05 — COMMENTED** [review]
Code LGTM & everything works fine now!

I have just one minor question regarding the unit tests, thanks for improving them!

File: bundles/org.openhab.ui/web/src/pages/settings/pages/sitemap/__tests__/sitemap-edit_jest.spec.js (hunk 1 above)
Comment (mherwege) [reply bundles/org.openhab.ui/web/src/pages/settings/pages/sitemap/__tests__/sitemap-edit_jest.spec.js]: It shouldn't have been removed. I don't know what happened there. I put it back in.

**Review by florian-h05 — APPROVED** [review]
LGTM, thanks!
